### PR TITLE
STAC-14474 Adds clarifications to documentation for Kubernetes backup/restore

### PR DESCRIPTION
**Pull request by vpartington**
See https://stackstate.atlassian.net/browse/STAC-14474

### CODE REVIEW

File: setup/data-management/backup_restore/kubernetes_backup.md [hunk 1]
```diff
@@ -111,6 +111,19 @@ The StackGraph and Elasticsearch backups are stored in BLOB containers called `s
...
 ### Backup to Kubernetes storage
 
+{% hint style="warning" %}
+If MinIO is configured to store its data in Kubernetes storage, a PersistentVolumeClaim (PVC) is used to request storage from the Kubernetes cluster. The kind of storage that is allocated depends on the configuration of the cluster.
+
+Please take the following considerations into account:
+
+1. Kubernetes clusters running in a cloud provider usually map PVCs to block storage, such as Elastic Block Storage for AWS or Azure Block Storage. Block storage is expensive, especially for large data volumes.
+2. Persistent Volumes are destroyed when the cluster that created them is destroyed. That means an (accidental) deletion of your cluster will also destroy all backups stored in Persistent Volumes.
+3. Persistent Volumes can not be accessed from another cluster. That means that it is not possible to restore StackState from a backup taken on another cluster.
```
Comment (gggina) [inline setup/data-management/backup_restore/kubernetes_backup.md:121]: "can not"
should be
"cannot"

File: setup/data-management/backup_restore/kubernetes_backup.md [hunk 2]
```diff
@@ -111,6 +111,19 @@ The StackGraph and Elasticsearch backups are stored in BLOB containers called `s
...
+
+Therefore, it is advised to use AWS S3 for clusters running on Amazon AWS and Azure Blob Storage for clusters running on Azure.
```
Comment (gggina) [inline setup/data-management/backup_restore/kubernetes_backup.md:123]: I would put this info above the numbered list in place of the line "please take note...."

e.g.
"It is advised to use AWS S3 for clusters running on Amazon AWS and Azure Blob Storage for clusters running on Azure for the following reasons:"

File: setup/data-management/backup_restore/kubernetes_backup.md [hunk 3]
```diff
@@ -196,10 +209,11 @@ Scripts to list and restore backups and snapshots can be found in the [restore d
...
 1. The `kubectl` binary has been installed.
-2. The `kubectl` binary is configured to connect to the Kubernetes cluster and the namespace within that cluster that runs StackState.
-3. The Helm value `backup.enabled` is set to `true`.
-4. The Helm value `backup.stackGraph.restore.enabled` is not set to `false` \(to access StackGraph backups\).
-5. The Helm value `backup.elasticsearch.restore.enabled` is not set to `false` \(to access Elasticsearch snapshots\).
+2. The `kubectl` binary is configured to connect to the Kubernetes cluster where StackState has been installed.
+3. The `kubectl` binary is configured to connect to the namespace within that cluster where StackState has been installed.
+4. The Helm value `backup.enabled` is set to `true`.
+5. The Helm value `backup.stackGraph.restore.enabled` is not set to `false` \(to access StackGraph backups\).
+6. The Helm value `backup.elasticsearch.restore.enabled` is not set to `false` \(to access Elasticsearch snapshots\).
```
Comment (gggina) [inline setup/data-management/backup_restore/kubernetes_backup.md:216]: grouping these would prevent the same words being repeated at the beginning of each line:

1. The `kubectl` binary is installed and configured to connect to:
    a. the Kubernetes cluster where StackState has been installed.
    b. the namespace within that cluster where StackState has been installed.
2. The following Helm values have been correctly set:
    a. `backup.enabled` is set to `true`.
    b. `backup.stackGraph.restore.enabled` is not set to `false` \(required to access StackGraph backups\).
    c. `backup.elasticsearch.restore.enabled` is not set to `false` \(required to access Elasticsearch snapshots\).

File: setup/data-management/backup_restore/kubernetes_backup.md [hunk 4]
```diff
@@ -52,7 +52,7 @@ minio:
 
 Replace the following values:
 
-* `YOUR_ACCESS_KEY` and `YOUR_SECRET_KEY` are the credentials that will be used to secure the MinIO system. The automatic backup jobs and the restore jobs will use them. They are also required to manually access the MinIO storage. `YOUR_ACCESS_KEY` should contain 5 to 20 alphanumerical characters and `YOUR_SECRET_KEY` should contain 8 to 40 alphanumerical characters.
+* `YOUR_ACCESS_KEY` and `YOUR_SECRET_KEY` are the credentials that will be used to secure the MinIO system. These credentials can be set to any valid value; `YOUR_ACCESS_KEY` should contain 5 to 20 alphanumerical characters and `YOUR_SECRET_KEY` should contain 8 to 40 alphanumerical characters. These credentials are set on the MinIO system and used by the automatic backup jobs and the restore jobs. They are also required if you want to manually access the MinIO system.
```
Comment (gggina) [inline setup/data-management/backup_restore/kubernetes_backup.md:55]: * `YOUR_ACCESS_KEY` and `YOUR_SECRET_KEY` are the credentials that will be used to secure the MinIO system. These credentials are set on the MinIO system and used by the automatic backup jobs and the restore jobs. They are also required if you want to manually access the MinIO system.
     * `YOUR_ACCESS_KEY` should contain 5 to 20 alphanumerical characters.
     * `YOUR_SECRET_KEY` should contain 8 to 40 alphanumerical characters.

File: setup/data-management/backup_restore/kubernetes_backup.md (hunk 1 above)
Comment (vpartington) [reply setup/data-management/backup_restore/kubernetes_backup.md:121]: Urgh! Thanks for catching that.

File: setup/data-management/backup_restore/kubernetes_backup.md (hunk 2 above)
Comment (vpartington) [reply setup/data-management/backup_restore/kubernetes_backup.md:123]: Good idea.

File: setup/data-management/backup_restore/kubernetes_backup.md (hunk 3 above)
Comment (vpartington) [reply setup/data-management/backup_restore/kubernetes_backup.md:216]: Got it, thanks!

File: setup/data-management/backup_restore/kubernetes_backup.md (hunk 4 above)
Comment (vpartington) [reply setup/data-management/backup_restore/kubernetes_backup.md:55]: 👍

**Review by gggina — APPROVED** [review]

**Pull request merged**